tests/MongoDB.Driver.Tests/JsonDrivenTests/JsonDrivenTestRunnerTest.cs

```diff
@@ -36,9 +37,9 @@ protected JsonDrivenTestRunnerTest(IJsonDrivenTestRunner testRunner, Dictionary<
 
         protected IJsonDrivenTestRunner TestRunner => _testRunner;
 
-        protected IServer GetPinnedServer()
+        protected EndPoint GetPinnedServerEndpoint()
```

Comment:
we need to get an endpoint from the session (rather than the whole server) and then get the required server based on the endpoint value from the specialized client